bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -328,18 +328,23 @@ protected synchronized void setRule(String rUID) {
             errMsgs = errMsgs + "\n" + errMessage;
         }
 
-        try {
-            ConnectionValidator.validateConnections(r);
-        } catch (Exception e) {
-            errMsgs = errMsgs + "\n Validation of rule" + r.getUID() + "has failed! " + e.getMessage();
-        }
-
         if (errMsgs == null) {
-            resolveDefaultValues(r);
-            register(r);
 
-            // change state to IDLE
-            setRuleStatusInfo(r.getUID(), new RuleStatusInfo(RuleStatus.IDLE));
+            try {
+                ConnectionValidator.validateConnections(r);
+                resolveDefaultValues(r);
+                register(r);
+                // change state to IDLE
+                setRuleStatusInfo(r.getUID(), new RuleStatusInfo(RuleStatus.IDLE));
+
+            } catch (IllegalArgumentException e) {
```

Comment:
But in this case the other code has to check again if there is an error message. 
this code:
  resolveDefaultValues(r);
  register(r);
 setRuleStatusInfo(r.getUID(), new RuleStatusInfo(RuleStatus.IDLE));

must not be called when the validation is faild.